packages/vscode-extension/src/project/devtools.ts

```diff
@@ -261,7 +274,9 @@ export class CDPDevtoolsServer extends DevtoolsServer implements Disposable {
       shutdownListener.dispose();
     });
     connection.onDisconnected(() => {
-      this.setConnection(undefined);
+      if (this.connection === connection) {
+        this.setConnection(undefined);
+      }
```

Comment:
I don't think this check can ever fail?
It's still fine to leave it as is, of course, since we might change things around it in the future.

Reply:
wanted to add it here since session is actually controlled by the superclass. One thing I was wondering was whether the resetting connection logic shouldn't be a part of the superclass, but was too afraid to change this bit